scipy/optimize/shgo_m/sobol_seq.py

```diff
@@ -0,0 +1,371 @@
+"""
+  Licensing:
+    This code is distributed under the MIT license.
+
+
+  Authors:
+    Original FORTRAN77 version of i4_sobol by Bennett Fox.
+    MATLAB version by John Burkardt.
+    PYTHON version by Corrado Chisari
+
+    Original Python version of is_prime by Corrado Chisari
+
+    Original MATLAB versions of other functions by John Burkardt.
+    PYTHON versions by Corrado Chisari
+
+    Original code is available from
+    http://people.sc.fsu.edu/~jburkardt/py_src/sobol/sobol.html
+
+  Modifications:
+    Wrapped into Python class [30.10.2017]
+"""
+import numpy as np
+
+__all__ = ['Sobol']
+
+class Sobol:
+    def __init__(self):
+        # Init class variables
+        self.atmost = None
+        self.dim_max = None
+        self.dim_num_save = None
+        self.initialized = None
+        self.lastq = None
+        self.log_max = None
+        self.maxcol = None
+        self.poly = None
+        self.recipd = None
+        self.seed_save = None
+        self.v = None
+
+    def i4_sobol_generate(self, dim_num, n, skip=1):
+        """
+        i4_sobol_generate generates a Sobol dataset.
+
+        Parameters:
+          Input, integer dim_num, the spatial dimension.
+          Input, integer N, the number of points to generate.
+          Input, integer SKIP, the number of initial points to skip.
+
+          Output, real R(M,N), the points.
+        """
+        r = np.full((n, dim_num), np.nan)
+        for j in range(n):
+            seed = j + skip
+            r[j, 0:dim_num], next_seed = self.i4_sobol(dim_num, seed)
+
+        return r
+
+    def i4_bit_hi1(self, n):
```

Comment:
In Python this function is already available from the integers 

```python
>>> (1023).bit_length
10
>>> (1025).bit_length
11
```

I don't know the range of numbers but 
```
def i4_bit_hi1(self, n):
    return int(x).bit_length()
```
and some positivity checks would be sufficient